external-crates/move/crates/move-analyzer/trace-adapter/src/runtime.ts

```diff
@@ -433,13 +433,25 @@ export class Runtime extends EventEmitter {
         // local variable array
         const frameLocalLifetimeEnds = this.trace.localLifetimeEnds.get(currentFrame.id);
         if (frameLocalLifetimeEnds) {
-            for (let i = 0; i < currentFrame.locals.length; i++) {
+            const localsLength = currentFrame.locals.length;
+            for (let i = 0; i < localsLength; i++) {
                 for (let j = 0; j < currentFrame.locals[i].length; j++) {
                     if (frameLocalLifetimeEnds[j] === instructionEvent.pc) {
                         currentFrame.locals[i][j] = undefined;
                     }
                 }
             }
+            // trim shadowed scopes that have no live variables in them
+            if (localsLength > 1) {
```

Comment:
I am not sure I understand why we must have at least on local in the frame to enter the loop

Reply:
Indeed there is not need for this if condition. Removed!